src/Common/src/Interop/OSX/Interop.libproc.cs

```diff
@@ -0,0 +1,523 @@
+        {
+            // Negative PIDs are invalid
+            if (pid < 0)
+            {
+                throw new ArgumentOutOfRangeException("pid", SR.NegativePidNotSupported);
+            }
+
+            bool success = false;
+
+            // Allocate our unmanaged buffer for the task info
+            info = default(T);
+            int size = Marshal.SizeOf<T>();
+            IntPtr buffer = Marshal.AllocCoTaskMem(size);
+
+            // Get the process information for the given PID
+            int result = proc_pidinfo(pid, flavor, args, buffer, size);
+            if (result == size)
+            {
+                // Copy the struct from unmanaged memory to managed memory
+                info = Marshal.PtrToStructure<T>(buffer);
+                success = true;
+            }
+
+            // Free our unmanaged buffer
+            Marshal.FreeCoTaskMem(buffer);
+
+            return success;
+        }
+
+        /// <summary>
+        /// Gets the full path to the executable file identified by the specified PID
+        /// </summary>
+        /// <param name="pid">The PID of the running process</param>
+        /// <param name="buffer">A pointer to an allocated block of memory that will be filled with the process path</param>
+        /// <param name="bufferSize">The size of the buffer, should be PROC_PIDPATHINFO_MAXSIZE</param>
+        /// <returns>Returns the length of the path returned on success</returns>
+        [DllImport(Interop.Libraries.libproc, CharSet = CharSet.Ansi, SetLastError = true)]
+        private static extern int proc_pidpath(
+            int pid, 
+            IntPtr buffer, 
+            uint bufferSize);
+
+        /// <summary>
+        /// Gets the full path to the executable file identified by the specified PID
+        /// </summary>
+        /// <param name="pid">The PID of the running process</param>
+        /// <returns>Returns the full path to the process executable</returns>
+        internal static string proc_pidpath(int pid)
+        {
+            // Negative PIDs are invalid
+            if (pid < 0)
+            {
+                throw new ArgumentOutOfRangeException("pid", SR.NegativePidNotSupported);
+            }
+
+            // Allocate our unmanaged buffer for the path
+            string path = string.Empty;
+            uint size = (uint)Marshal.SizeOf<char>() * PROC_PIDPATHINFO_MAXSIZE + 1;
+            IntPtr buffer = Marshal.AllocCoTaskMem((int)size);
+
+            // Get the PID's executable's path
+            int result = proc_pidpath(pid, buffer, size);
+            if (result < 0)
+            {
+                // Check for less than 0 since the kernel process won't have a path
+                throw new COMException(string.Format(SR.CantFindProcessExecutablePath, pid), Marshal.GetLastWin32Error());
+            }
+
+            // Copy unmanaged data into a managed string and free the unmanaged buffer
+            path = Marshal.PtrToStringUni(buffer, result);
```

Comment:
Just double-checking that "Uni" here is correct and that the data coming back from the OS is in fact UTF16 encoded?